### PR TITLE
CLOUDP-47099: Snyk snapshot on Evergreen

### PR DESCRIPTION
**Pull request by fabianlindfors**
We are using Snyk to track dependencies and find vulnerabilities. With this PR we add a new task to Evergreen which will publish a snapshot to Snyk. Snyk will analyze the snapshot and report any vulnerabilities asynchronously. It will also keep track of any new vulnerabilities that pop up. This test should always pass.

I also took the opportunity to refactor our Evergreen file with a new variant `code_health`. Previously `go vet` was run as part of the unit test which didn't make much sense. Now vet and Snyk are run as part of the code_health variant.

### CODE REVIEW

File: .evergreen.yaml [hunk 1]
```diff
@@ -20,6 +20,22 @@ functions:
...
         script: |
           go vet ./...
+  "setup_snyk":
+    - command: shell.exec
+      type: system
+      params:
+        script: |
+          curl -L -o snyk https://github.com/snyk/snyk/releases/download/v1.216.0/snyk-linux
+          chmod +x snyk
+          mv snyk /usr/local/bin/
```
Comment (jamesbroadhead) [inline .evergreen.yaml]: please don't do this -- all evergreen file access should be within the workspace dir that the tasks run in. The machines run many tasks before being shut down, so changes to the root filesystem will affect future runs.

Instead: 
`````mkdir bin
export PATH="$(pwd)/bin:$PATH"
```

File: .evergreen.yaml [hunk 2]
```diff
@@ -20,6 +20,22 @@ functions:
...
+  "snyk_snapshot":
+    - command: shell.exec
+      type: system
+      params:
+        working_dir: src/atlas-service-broker
+        script: |
+          snyk auth ${snyk_token}
```
Comment (jamesbroadhead) [inline .evergreen.yaml]: Added this as "snyk-evergreen-upload-atlas-service-broker" 

It already had this name, so I wanted to be consistent.

File: .evergreen.yaml (hunk 1 above)
Comment (fabianlindfors) [reply .evergreen.yaml]: That's a very good point, thanks! Fixed now.

File: .evergreen.yaml (hunk 2 above)
Comment (fabianlindfors) [reply .evergreen.yaml]: Sorry, what are you referring to with "snyk-evergreen-upload-atlas-service-broker"?

Comment (fabianlindfors) [reply .evergreen.yaml]: Updated the name of the expansion to "snyk-evergreen-upload-atlas-service-broker".

**Comment by fabianlindfors** [comment]
The Snyk task will output a URL for the snapshot which is one way to find it in Evergreen. It can also be viewed by finding the project in Snyk: https://app.snyk.io/org/mongodb-default/project/3e49f5bb-6275-403e-aea1-fe2cfe1f151f.

**Pull request merged**